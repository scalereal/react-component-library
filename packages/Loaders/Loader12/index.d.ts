import * as React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
declare const Loader12: React.FC<Props> & WithStyle;
export default Loader12;
