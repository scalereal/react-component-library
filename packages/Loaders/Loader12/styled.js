var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
export var Loader = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    border-radius: 50%;\n\tbox-shadow: 0 1em 0 -.2em currentcolor;\n\tposition: relative;\n\tanimation: loader-14 0.8s ease-in-out alternate infinite;\n\tanimation-delay: 0.32s;\n\ttop: -1em;\n\t&:after,\n\t&:before {\n\t\tcontent: '';\n\t\tposition: absolute;\n\t\twidth: inherit;\n\t\theight: inherit;\n\t\tborder-radius: inherit;\n\t\tbox-shadow: inherit;\n\t\tanimation: inherit;\n\t}\n\t&:before {\n\t\tleft: -1em;\n\t\tanimation-delay: 0.48s;\n\t}\n\t&:after {\n\t\tright: -1em;\n\t\tanimation-delay: 0.16s;\n\t}\n\n    @keyframes loader-14 {\n        0% {\n            box-shadow: 0 2em 0 -.2em currentcolor;\n        }\n        100% {\n            box-shadow: 0 1em 0 -.2em currentcolor;\n        }\n    }\n"], ["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    border-radius: 50%;\n\tbox-shadow: 0 1em 0 -.2em currentcolor;\n\tposition: relative;\n\tanimation: loader-14 0.8s ease-in-out alternate infinite;\n\tanimation-delay: 0.32s;\n\ttop: -1em;\n\t&:after,\n\t&:before {\n\t\tcontent: '';\n\t\tposition: absolute;\n\t\twidth: inherit;\n\t\theight: inherit;\n\t\tborder-radius: inherit;\n\t\tbox-shadow: inherit;\n\t\tanimation: inherit;\n\t}\n\t&:before {\n\t\tleft: -1em;\n\t\tanimation-delay: 0.48s;\n\t}\n\t&:after {\n\t\tright: -1em;\n\t\tanimation-delay: 0.16s;\n\t}\n\n    @keyframes loader-14 {\n        0% {\n            box-shadow: 0 2em 0 -.2em currentcolor;\n        }\n        100% {\n            box-shadow: 0 1em 0 -.2em currentcolor;\n        }\n    }\n"])), function (pr) { return pr.Color ? pr.Color : 'inherit'; });
var mixin = function (size) {
    switch (size) {
        case "XXS":
            return "font-size: 0.8rem;\n            " + Loader + "{\n                margin: 0 15px 15px 15px;\n                top: -1em;\n            }\n           ";
        case "XS":
            return "font-size: 1rem;\n            " + Loader + "{\n                margin: 0 15px 15px 15px;\n                top: -1em;\n            }\n            \n            ";
        case "S":
            return "font-size: 2rem;\n            " + Loader + "{\n                margin: 0 25px 25px 25px;\n                top: -1.1em;\n            }\n            ";
        case "M":
            return "font-size: 3rem;\n            " + Loader + "{\n                margin: 0 40px 35px 40px;\n                top: -1.1em;\n            }\n            ";
        case "L":
            return "font-size: 5rem;\n            " + Loader + "{\n                margin: 0 65px 50px 65px;\n                top: -1.2em;\n            }\n            ";
        case "XL":
            return "font-size: 6rem;\n            " + Loader + "{\n                margin: 0 80px 60px 80px;\n                top: -1.2em;\n            }\n            ";
        default:
            return "font-size: 1rem;\n        " + Loader + "{\n            margin:0 25px;\n        }\n       ";
    }
};
export var LoaderWrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    ", "\n    ", "\n    display: ", ";\n"], ["\n    ", "\n    ", "\n    display: ", ";\n"])), function (pr) { return pr.disabled === true ? 'pointer-events: none;opacity: 0.4;' : ''; }, function (pr) { return mixin(pr.size); }, function (pr) { return pr.visible ? 'inline-block' : 'none'; });
var templateObject_1, templateObject_2;
