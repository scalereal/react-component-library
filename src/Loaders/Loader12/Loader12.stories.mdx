import Loader12 from '.';
import { SizeMap } from '../types';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/Loader12"
    component={Loader12}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader12 {...args} />
    </div>
);

<Preview>
    <Story name="Loader12" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexWrap: 'wrap' }}>
        <Loader12 size="S" Color="green" />
        <Loader12 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexDirection: 'column' }}>
        <Loader12 size="XXS" />
        <Loader12 size="XS" />
        <Loader12 size="S" />
        <Loader12 size="M" />
        <Loader12 size="L" />
        <Loader12 size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader12" />
