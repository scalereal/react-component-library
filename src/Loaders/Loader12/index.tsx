import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

const Loader12:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

Loader12.displayName="Loader12";
Loader12.defaultProps = {
    size:'S',
    display: true
};

export default Loader12;
