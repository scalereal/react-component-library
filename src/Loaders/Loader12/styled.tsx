import styled from 'styled-components';
import { Props } from '../types';

export const Loader = styled.div<Props>`
    display: inline-block;
    width: 1em;
    height: 1em;
    color: ${pr=>pr.Color?pr.Color:'inherit'};
    vertical-align: middle;
    pointer-events: none;
    
    border-radius: 50%;
	box-shadow: 0 1em 0 -.2em currentcolor;
	position: relative;
	animation: loader-14 0.8s ease-in-out alternate infinite;
	animation-delay: 0.32s;
	top: -1em;
	&:after,
	&:before {
		content: '';
		position: absolute;
		width: inherit;
		height: inherit;
		border-radius: inherit;
		box-shadow: inherit;
		animation: inherit;
	}
	&:before {
		left: -1em;
		animation-delay: 0.48s;
	}
	&:after {
		right: -1em;
		animation-delay: 0.16s;
	}

    @keyframes loader-14 {
        0% {
            box-shadow: 0 2em 0 -.2em currentcolor;
        }
        100% {
            box-shadow: 0 1em 0 -.2em currentcolor;
        }
    }
`;

const mixin = (size:any) => {
    switch (size) {
        case "XXS":
            return `font-size: 0.8rem;
            ${Loader}{
                margin: 0 15px 15px 15px;
                top: -1em;
            }
           `;
        case "XS":
            return `font-size: 1rem;
            ${Loader}{
                margin: 0 15px 15px 15px;
                top: -1em;
            }
            
            `;
        case "S":
            return `font-size: 2rem;
            ${Loader}{
                margin: 0 25px 25px 25px;
                top: -1.1em;
            }
            `;
        
        case "M":
            return `font-size: 3rem;
            ${Loader}{
                margin: 0 40px 35px 40px;
                top: -1.1em;
            }
            `;

        case "L":
            return `font-size: 5rem;
            ${Loader}{
                margin: 0 65px 50px 65px;
                top: -1.2em;
            }
            `;
        
        case "XL":
            return `font-size: 6rem;
            ${Loader}{
                margin: 0 80px 60px 80px;
                top: -1.2em;
            }
            `;
  
      default:
        return `font-size: 1rem;
        ${Loader}{
            margin:0 25px;
        }
       `;
     }
 };

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    display: ${pr=>pr.visible?'inline-block':'none'};
`;
