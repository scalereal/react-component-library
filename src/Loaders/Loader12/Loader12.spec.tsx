import { render } from '@testing-library/react';
import * as React from 'react';
import Loader12 from '.';

describe('Loader12 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<Loader12/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <Loader12
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <Loader12 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});