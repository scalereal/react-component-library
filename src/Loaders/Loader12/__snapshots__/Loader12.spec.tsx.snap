// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader12 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz fVkNb"
  >
    <div
      class="sc-bdfBwQ duuDkO"
    />
  </div>
</div>
`;

exports[`Loader12 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz ciATyB"
  >
    <div
      class="sc-bdfBwQ bfKKDX"
    />
  </div>
</div>
`;

exports[`Loader12 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz xBROB"
  >
    <div
      class="sc-bdfBwQ duuDkO"
    />
  </div>
</div>
`;
